### PR TITLE
Add agent plugin to support Sentinel

### PR DESCRIPTION
**Pull request by liqiangz**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Is this a plugin Sentinel SDK?

**Comment by liqiangz** [comment]
> Is this a plugin Sentinel SDK?

Yes. https://github.com/alibaba/Sentinel

File: apm-sniffer/apm-sdk-plugin/sentinel-1.x-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <url>http://maven.apache.org</url>
+    <properties>
+        <sentinel-core.version>1.7.2</sentinel-core.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.alibaba.csp</groupId>
+            <artifactId>sentinel-core</artifactId>
+            <version>${sentinel-core.version}</version>
```
Comment (kezhenxu94) [inline apm-sniffer/apm-sdk-plugin/sentinel-1.x-plugin/pom.xml:40]: Should be provided scope

**Comment by wu-sheng** [comment]
`Sentinel` has only 31 commits in the last 5 months, and the last release is on 4 Feb. Is the project in low maintenance mode? Or it becomes low activity for some reason?

**Comment by wu-sheng** [comment]
Check the data from here, https://github.com/alibaba/Sentinel/releases/tag/1.8.1

**Comment by liqiangz** [comment]
> `Sentinel` has only 31 commits in the last 5 months, and the last release is on 4 Feb. Is the project in low maintenance mode? Or it becomes low activity for some reason?

Sentinel is the default flow control component of spring cloud alibaba, and it is widely used now. You can ask @zhaoyuguang  about project updates. He is the commiter of alibaba/sentinel.

**Comment by wu-sheng** [comment]
> > `Sentinel` has only 31 commits in the last 5 months, and the last release is on 4 Feb. Is the project in low maintenance mode? Or it becomes low activity for some reason?
> 
> Sentinel is the default flow control component of spring cloud alibaba(https://spring.io/projects/spring-cloud-alibaba), and it is widely used now. You can ask @zhaoyuguang about project updates. He is the commiter of alibaba/sentinel.

@zhaoyuguang Any update? 7 commits per month seems very low.

File: apm-sniffer/apm-sdk-plugin/sentinel-1.x-plugin/pom.xml (hunk 1 above)
Comment (liqiangz) [reply apm-sniffer/apm-sdk-plugin/sentinel-1.x-plugin/pom.xml:40]: Done, thanks.

**Comment by zhaoyuguang** [comment]
> > > `Sentinel` has only 31 commits in the last 5 months, and the last release is on 4 Feb. Is the project in low maintenance mode? Or it becomes low activity for some reason?
> > 
> > 
> > Sentinel is the default flow control component of spring cloud alibaba(https://spring.io/projects/spring-cloud-alibaba), and it is widely used now. You can ask @zhaoyuguang about project updates. He is the commiter of alibaba/sentinel.
> 
> @zhaoyuguang Any update? 7 commits per month seems very low.

Sentinel is updated all the time, and will release version 1.8.2 in the near future, but the commits count are relatively small.

**Comment by zhaoyuguang** [comment]
@liqiangz Sentinel has a large number of users. With the introduction of Sentinel, the application service will generate a large number of LocalSpan as currently designed. From this point of view, is it better to put it in an optional plug-in?

**Comment by liqiangz** [comment]
> @liqiangz Sentinel has a large number of users. With the introduction of Sentinel, the application service will generate a large number of LocalSpan as currently designed. From this point of view, is it better to put it in an optional plug-in?

Because the similar hystrix plugin is now placed in the default, so I put it in the default plugin. But what you said also makes sense,  it can be put in optional plugins.

**Comment by zhaoyuguang** [comment]
> > @liqiangz Sentinel has a large number of users. With the introduction of Sentinel, the application service will generate a large number of LocalSpan as currently designed. From this point of view, is it better to put it in an optional plug-in?
> 
> Because the similar hystrix plugin is now placed in the default, so I put it in the default plugin. But what you said also makes sense, it better be put in optional plugins.

Yes, the Hystrix Plugin does produce a lot of LocalSpans, but it's a little different than Sentinel

- Not import by the Hystrix Plugin will result in trace chain breakage, which Sentinel will not breakage, because Hystrix mostly implements isolation for thread pools

- Hystrix requires secondary development by developers, but Sentinel is compatible with almost every framework, developers just import Sentinel in pom.xml

**Comment by wu-sheng** [comment]
Please fix the muse detected issue.

**Comment by liqiangz** [comment]
> Please fix the muse detected issue.

Could I ignore the current issue?  I have upgraded the spring-boot to the latest version, and this issue is just a test code.

File: test/plugin/scenarios/sentinel-scenario/pom.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                <artifactId>spring-boot-dependencies</artifactId>
+                <version>${spring-boot-version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply test/plugin/scenarios/sentinel-scenario/pom.xml:51]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (wu-sheng) [reply test/plugin/scenarios/sentinel-scenario/pom.xml:51]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (wu-sheng) [reply test/plugin/scenarios/sentinel-scenario/pom.xml:51]: @muse-dev ignore

[the remark this answers is not in the document]
Comment (wu-sheng) [reply test/plugin/scenarios/sentinel-scenario/pom.xml:51]: @muse-dev ignore

**Comment by wu-sheng** [comment]
@zhaoyuguang As an official committer of that project, could you check the plugin? I think you should be qualified and official.

**Review by zhaoyuguang — APPROVED** [review]
LGTM

**Review by wu-sheng — DISMISSED** [review]
@zhaoyuguang approved, I just forward it.

File: docs/en/setup/service-agent/java-agent/README.md [hunk 3]
```diff
@@ -193,6 +193,7 @@ Now, we have the following known optional plugins.
 * Plugin of quartz-scheduler-2.x in the optional plugin folder. The reason for being an optional plugin is, many task scheduling systems are based on quartz-scheduler, this will cause duplicate tracing and link different sub-tasks as they share the same quartz level trigger, such as ElasticJob.
 * Plugin of spring-webflux-5.x in the optional plugin folder. Please only activate this plugin when you use webflux alone as a web container. If you are using SpringMVC 5 or Spring Gateway, you don't need this plugin.
 * Plugin of mybatis-3.x in optional plugin folder. The reason of being optional plugin is, many local span are generated, which also spend more CPU, memory and network.
+* Plugin of sentinel-1.x in optional plugin folder. The reason of being optional plugin is, sentinel will generate a large number of LocalSpan as currently designed.
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/README.md:196]:
```suggestion
* Plugin of sentinel-1.x in the optional plugin folder. The reason for being an optional plugin is, the sentinel plugin generates a large number of local spans, which have a potential performance impact.
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**